### PR TITLE
fix: allow parameters to be passed through to `SamTranslatorWrapper`

### PR DESCRIPTION
**Pull request by sriram-mv**
Why is this change necessary?

* Some plugins require that intrinsics are resolved and parameter
information be present.

How does it address the issue?

* Passing parameters that have parameters and pseudo parameters resolved
fixes the issue.
* Also tested with local invoke and deploy

What side effects does this change have?

* None at this point.

*Issue #, if available:*

* https://github.com/awslabs/aws-sam-cli/issues/1958
* https://github.com/awslabs/aws-sam-cli/issues/1956

*Why is this change necessary?*

*How does it address the issue?*

*What side effects does this change have?*

*Did you change a dependency in `requirements/base.txt`?*
    *If so, did you run `make update-reproducible-reqs`*

*Checklist:*

- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
- [x] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by sriram-mv** [comment]
Yes, integ tests will be added.

File: tests/integration/local/invoke/test_integrations_cli.py [hunk 1]
```diff
@@ -401,9 +401,9 @@ def test_skip_pull_image_in_env_var(self):
         self.assertIn("Requested to skip pulling images", process_stderr.decode("utf-8"))
 
     @skipIf(SKIP_LAYERS_TESTS, "Skip layers tests in Appveyor only")
-    @skipIf(IS_WINDOWS, "This test failes on windows due to unix permissions not set properly on unzipped binary")
+    @skipIf(IS_WINDOWS, "This test fails on windows due to unix permissions not set properly on unzipped binary")
     @pytest.mark.flaky(reruns=3)
-    def test_invoke_returns_execpted_results_from_git_function(self):
+    def test_invoke_returns_expected_results_from_git_function(self):
```
Comment (jfuss) [inline tests/integration/local/invoke/test_integrations_cli.py:406]: Thanks for fixing my spelling errors.

File: tests/integration/testdata/invoke/template-pseudo-params.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: Run intrinsics and fill in pseudo parameters first.
+
+Mappings:
+  MappingExample:
+    eu-west-1:
+      ApplicationId: arn:aws:serverlessrepo:us-east-1:077246666028:applications/hello-world
```
Comment (jfuss) [inline tests/integration/testdata/invoke/template-pseudo-params.yaml:8]: Can we use the `1234...` fake account here?

Comment (sriram-mv) [reply tests/integration/testdata/invoke/template-pseudo-params.yaml:8]: sure

Comment (sriram-mv) [reply tests/integration/testdata/invoke/template-pseudo-params.yaml:8]: turns out we cant, the serverless app plugin makes a call to serverless application repo based on the applicationId.

**Pull request merged**